pkg/cmd/roachtest/tests/large_schema_backup_restore.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/stretchr/testify/require"
+)
+
+// registerLargeSchemaBackupRestores registers tests that large schemas can
+// backup/restore successfully.
+func registerLargeSchemaBackupRestores(r registry.Registry) {
+	// 10k is probably the upper limit right now, but we haven't tested further.
+	largeSchemaBackupRestore(r, 10000)
+}
+
+func largeSchemaBackupRestore(r registry.Registry, numTables int) {
+	r.Add(registry.TestSpec{
+		Name:      fmt.Sprintf("backup-restore/large-schema/tables=%d", numTables),
+		Owner:     registry.OwnerDisasterRecovery,
+		Benchmark: false,
+		Cluster: r.MakeClusterSpec(
+			// The roachprod default for GCE and the backup testing bucket are both us-east1,
+			// so no special region assignment is required
+			9,
+			spec.CPU(8),
+			spec.GCEMachineType("n2-standard-8"),
+		),
+		CompatibleClouds: registry.OnlyGCE,
+		Suites:           registry.Suites(registry.Weekly),
+		Timeout:          12 * time.Hour,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			// Number of tables per-database from the TPCC template.
+			const numTablesPerDB = 100
+			numDbs := numTables/numTablesPerDB + 1
+			dbs := make([]string, numDbs)
+			tables := make([]string, numTables)
+
+			c.Start(ctx, t.L(), option.NewStartOpts(option.NoBackupSchedule), install.MakeClusterSettings(), c.CRDBNodes())
+			conn := c.Conn(ctx, t.L(), 1)
+			defer conn.Close()
+
+			// Create a user that can log in to the console
+			_, err := conn.ExecContext(ctx, "CREATE USER roachadmin password 'roacher'")
```

Comment:
i don't believe this is necessary? You can use username: roachprod; password: cockroachdb

Reply:
Hey, look at that! 👍